File: frontend/instruction-platform-client/src/instructionLabels.js
export const instructionCategories = [
  { value: 'FireSafety', label: 'Пожарная безопасность' },
  { value: 'ElectricalSafety', label: 'Электробезопасность' },
  { value: 'OccupationalSafety', label: 'Охрана труда' }
]

export const instructionTypes = [
  { value: 'Introductory', label: 'Вводный' },
  { value: 'Primary', label: 'Первичный на рабочем месте' },
  { value: 'Repeated', label: 'Повторный' },
  { value: 'Unscheduled', label: 'Внеплановый' },
  { value: 'Targeted', label: 'Целевой' }
]

export function categoryLabel(value) {
  return instructionCategories.find((item) => item.value === value)?.label || value
}

export function instructionTypeLabel(value) {
  return instructionTypes.find((item) => item.value === value)?.label || value
}

export function isRetrainingOverdue(dueAt) {
  if (!dueAt) return false
  return new Date(dueAt) < new Date()
}

export function formatDate(value) {
  if (!value) return '—'
  return new Date(value).toLocaleDateString('ru-RU')
}

File: backend/InstructionPlatform.Api/Services/InstructionLabels.cs
using InstructionPlatform.Api.Domain.Enums;

namespace InstructionPlatform.Api.Services;

public static class InstructionLabels
{
    public static string Category(InstructionCategory category) => category switch
    {
        InstructionCategory.FireSafety => "Пожарная безопасность",
        InstructionCategory.ElectricalSafety => "Электробезопасность",
        InstructionCategory.OccupationalSafety => "Охрана труда",
        _ => category.ToString()
    };

    public static string Type(InstructionType type) => type switch
    {
        InstructionType.Introductory => "Вводный",
        InstructionType.Primary => "Первичный на рабочем месте",
        InstructionType.Repeated => "Повторный",
        InstructionType.Unscheduled => "Внеплановый",
        InstructionType.Targeted => "Целевой",
        _ => type.ToString()
    };
}
